pkg/services/dashboards/service/dashboard_service.go

```diff
@@ -1293,6 +1281,32 @@ func (dr *DashboardServiceImpl) FindDashboards(ctx context.Context, query *dashb
 	return dr.dashboardStore.FindDashboards(ctx, query)
 }
 
+func (dr *DashboardServiceImpl) fetchFolderNames(ctx context.Context, query *dashboards.FindPersistedDashboardsQuery, response *dashboardv0alpha1.SearchResults) (map[string]string, error) {
+	folderIds := []string{}
+	for _, hit := range response.Hits {
+		if hit.Folder != "" {
+			folderIds = append(folderIds, hit.Folder)
+		}
+	}
```

Comment:
Suggestion: we might want to do something like that 
```suggestion
	// Deduplicate folder IDs from search hits.
	folderSet := make(map[string]struct{}, len(response.Hits))
	for _, hit := range response.Hits {
		if hit.Folder != "" {
			folderSet[hit.Folder] = struct{}{}
		}
	}
```

And then use something like `UIDs: maps.Keys(folderSet)` in `GetFoldersQuery`. This way we make sure we don't have duplicated UIDs.

Reply:
good call @JohnnyQQQQ